### PR TITLE
Rt3 20 layoutsidenav

### PR DESCRIPTION
**Pull request by RobertBognar**

### CODE REVIEW

File: src/components/Layout/Layout.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import Navbar from '../Navbar/Navbar';
+import SideNav from '../Navbar/SideNav';
+
+//Hardcoded SideNav To Layout, Need To Fix With AuthContext
+const Layout = (props) => {
+    return (
+        <Fragment>
+            <Navbar />
+            <SideNav />
+            <main>{props.children}</main>
```
Comment (petarjs) [inline src/components/Layout/Layout.js]: Needs a bit more styling, currently it seems like `main` overlaps `SideNav` :))

image

File: src/App.js [hunk 2]
```diff
@@ -6,20 +6,24 @@ import Register from './components/Register';
...
-                    <Route path="/team" element={<HomepageAuthUser />} />
-                </Routes>
+                <Layout>
+                    <Routes>
+                        <Route path="/login" element={<Login />} />
+                        <Route path="/register" element={<Register />} />
+                        <Route path="/profile" element={<ProfilePage />} />
+                        <Route path="/board" element={<GuestHomepage />} />
+                    </Routes>
+                </Layout>
```
Comment (petarjs) [inline src/App.js]: I think we should do this a bit differently, we want to be able to specify the layout per route, or per route group - for example, for Login, Register and GuestHomepage components, we want to have a different layout component, e.g. GuestLayout, and for ProfilePage and other auth pages, we can have AuthLayout

Let's create a generic `<Page>` component that will have `layout` and `content` as properties, so we can have something like:

```jsx
<Route path="/board" element={<Page layout={GuestLayout} content={GuestHomepage} />
```

**Review by petarjs — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 2 above)
Comment (RobertBognar) [reply src/App.js]: I have tried with <Page> but couldn't manage to handle it arround. In new push I divided routes on GuestLayout & AuthLayout to group targeted routes.

Comment (RobertBognar) [reply src/App.js]: I think it's done now :)

Comment (RobertBognar) [reply src/App.js]: I think it is done now :)

**Comment by petarjs** [comment]
Deployment failed
image

You can see what is the error if you run `npm run build`

**Pull request merged**